### ISSUE TITLE
I have implemented a custom objective function. Do i still have to specify the objective parameter in param list or not.

### ISSUE THREAD

**Issue by atul-patel** [dmlc/xgboost#4736]
beta = 2
def weighted_logloss(y_hat,dtrain):
    y = dtrain.get_label()
    p = y_hat
    grad = (beta-1)*p*y + p - beta*y
    hess = ((beta-1)*y + 1)*(p * (1.0 - p))
    return grad,hess

def err_rate(y_hat, dtrain):
    y = dtrain.get_label()
    y_hat = np.clip(y_hat, 10e-7, 1-10e-7)
    loss_fn = beta*y*np.log(y_hat)
    loss_fp = (1.0 - y)*np.log(1.0 - y_hat)
    return 'log_loss', np.sum(-(loss_fn+loss_fp))/len(y)

param = {'max_depth': 9, 'learning_rate': 0.1,'objective': 'binary:logistic','min_child_weight' : 20,'min_split_loss' : 0,'reg_lambda' : 100,'tree_method' : 'exact','disable_default_eval_metric': 1}
param['nthread'] = 100

evallist = [(dtrain, 'train')]

num_round = 1000

bst = xgb.train(param, dtrain, num_round,obj = weighted_logloss,feval=err_rate,evals=[(dtrain, 'dtrain')])

**Comment by trivialfis** [comment]
@atul-patel See an example on master branch: https://xgboost.readthedocs.io/en/latest/tutorials/custom_metric_obj.html

**Comment by atul-patel** [comment]
I am aware of that tutorial. I just had a doubt how does the algorithm work when both the parameters are there. (i.e objective as well as obj). When using obj does the algo returns margin or after sigmoid predictions
